.editorconfig

```diff
@@ -0,0 +1,13 @@
+# EditorConfig info: https://editorconfig.org/
+root = true
+
+[Makefile*]
+indent_style = tab
+
+[*.{c,cpp,h}]
+indent_style = space
+indent_size = 4
+charset = utf-8
```

Comment:
I would have thought that 7-bit ASCII was a safer choice here. The common code needs to be portable to all possible targets. utf-8 does not seem conservative enough to me,

Reply:
Editorconfig only supports UTF-8, not ASCII, due to it being compatible in ASCII range, checking for non-ASCII characters should be rather done in some GitHub Action